packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -0,0 +1,310 @@
+import { useLocalCurrencyCode } from 'src/localCurrency/hooks'
+import { getLocalCurrencyExchangeRate } from 'src/localCurrency/selectors'
+import { emptyHeader, HeaderTitleWithBalance } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import { getRateForMakerToken, goldToDollarAmount } from 'src/utils/currencyExchange'
+
+const { decimalSeparator } = getNumberFormatSettings()
+
+type RouteProps = StackScreenProps<StackParamList, Screens.FiatExchangeAmount>
+
+type Props = RouteProps
+
+const oneDollarAmount = {
+  value: new BigNumber('1'),
+  currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code,
+}
+
+const useDollarAmount = (currency: CURRENCY_ENUM, amount: BigNumber) => {
+  const localExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencyCode = useLocalCurrencyCode()
+  const exchangeRatePair = useSelector(exchangeRatePairSelector)
+
+  if (currency === CURRENCY_ENUM.DOLLAR) {
+    return convertDollarsToMaxSupportedPrecision(
+      (!amount.isNaN() &&
+        convertLocalAmountToDollars(amount, localCurrencyCode ? localExchangeRate : 1)) ||
+        new BigNumber('0')
+    )
+  } else {
+    const exchangeRate = getRateForMakerToken(
+      exchangeRatePair,
+      CURRENCY_ENUM.DOLLAR,
+      CURRENCY_ENUM.GOLD
+    )
+    return goldToDollarAmount(amount, exchangeRate) || new BigNumber(0)
+  }
+}
+
+function FiatExchangeAmount({ route }: Props) {
+  const { t } = useTranslation(Namespaces.fiatExchangeFlow)
+
+  const [showingMinAmountDialog, setShowingMinAmountDialog] = useState(false)
+  const closeMinAmountDialog = () => setShowingMinAmountDialog(false)
+  const [showingDailyLimitDialog, setShowingDailyLimitDialog] = useState(false)
+
+  const [inputAmount, setInputAmount] = useState('')
+  const parsedInputAmount = parseInputAmount(inputAmount, decimalSeparator)
+  const exchangeRatePair = useSelector(exchangeRatePairSelector)
+  const localCurrencyExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencyCode = useLocalCurrencyCode()
+  const currencySymbol = LocalCurrencySymbol[localCurrencyCode]
+
+  const currency = route.params.currency
+  const isCusdCashIn = currency === CURRENCY_ENUM.DOLLAR
+  const inputCurrencySymbol = isCusdCashIn ? currencySymbol : ''
+
+  const dollarAmount = useDollarAmount(currency, parsedInputAmount)
+  const localCurrencyAmount = convertDollarsToLocalAmount(dollarAmount, localCurrencyExchangeRate)
+  const dailyLimitCusd = useSelector(cUsdDailyLimitSelector)
+  const minAmountInLocalCurrency = convertDollarsToLocalAmount(
+    DOLLAR_ADD_FUNDS_MIN_AMOUNT,
+    localCurrencyExchangeRate
+  )?.toFixed(0)
+
+  const dispatch = useDispatch()
+
+  React.useEffect(() => {
+    dispatch(fetchExchangeRate())
+  }, [])
+
+  function isNextButtonValid() {
```

Comment:
Can we get rid of this? Seems redundant if we also have a check elsewhere that displays an informative modal. Might be nice to just eliminate some code and in general, I think it's better UX to not disable buttons.

Reply:
It's just to make sure the value in the input is valid, it's not checking any values. The mockups have this invalid state and  I do think it's better UX to have it like this than allowing users to press the button and see an error.